planner/core/rule_column_pruning.go

```diff
@@ -129,6 +129,13 @@ func (ls *LogicalSort) PruneColumns(parentUsedCols []*expression.Column) {
 
 // PruneColumns implements LogicalPlan interface.
 func (p *LogicalUnionAll) PruneColumns(parentUsedCols []*expression.Column) {
+	used := getUsedList(parentUsedCols, p.schema)
+	for i := range used {
+		if used[i] {
```

Comment:
why append the unused columns into `parentUsedCols`?

Reply:
The used bool slice indicates whether the p.schema is used in parentUsedCols,
so we only need to append the unused columns. @zz-jason